### DISCUSSION TITLE
Username for matrix notification

### DISCUSSION THREAD

**Discussion by Eirikr70** [containrrr/watchtower#1400] [Q&A] [answered]
Hello, sorry for the poor question, but I have been trying to set up shoutrrr notifications from Watchtower to Matrix with all kind of username syntax, and it has always been rejected. My Matrix username is `@eric;mylastname.fr`, so I have tried `matrix://"@eric:mylastname.fr":Myverysecretpassword@192.168.1.104:8008/?rooms=Admin`. Since it has failed (`net/url: invalid userinfo`), I have tried with all kinds of combinations (`eric@mylastname.fr`, `eric`, `eric:mylastname.fr`, ...) with no more luck.
Can someone explain to me how I should 'spell' my username ?
Thanks a lot.

**Comment by Eirikr70** [comment, marked as answer]
Solved with tokens.

**Comment by piksel** [comment]
All parts of the URL need to be encoded if they have special characters. You can use the `encodeURIComponent` in the web browser console to do it (or using [this jsfiddle](https://jsfiddle.net/e17nmcgs/)). Your encoded username in the URL would be: `matrix://%40eric%3Amylastname.fr:Myverysecretpassword@192.168.1.104:8008`
